change branch/deopt to assume/checkpoint

skrynski opened the pull request.

skrynski committed:
change branch/deopt to assume/checkpoint

skrynski committed:
remove printing

skrynski committed:
removed DeoptReason

skrynski committed:
Merge branch 'master' into assumeNotBranch-rir2pir

skrynski committed:
first version

skrynski committed:
fix

skrynski committed:
minor fixes

skrynski committed:
minor fix

skrynski committed:
fix

skrynski committed:
tweaking logic

skrynski committed:
change to feedbackIsTrue

skrynski committed:
reduce variable scope

skrynski committed:
change deopt reason

o- commented:
Here are some stats for your PR:

* WARNING: the longest CI job test_features_3 took 1.94h
* real_thing suite improved by 1.03
* nbody_naive regressed by 0.78
* mandelbrot_naive_ascii regressed by 0.82
* fasta_3_strict regressed by 0.92
* vectorWhile regressed by 0.91
* Overall benchmarks improved by 1.0

Please find your performance results at https://rir-benchmarks.prl.fit.cvut.cz/diff?job_ids[]=1428292092&job_ids[]=1380012988&selection=all

skrynski committed:
Merge branch 'master' into assumeNotBranch-rir2pir

skrynski committed:
trying minor changes

o- commented:
Here are some stats for your PR:

* WARNING: the longest CI job tests_debug2 took 2.09h
* real_thing suite improved by 1.09
* convolution_slow regressed by 0.95
* nbody_naive regressed by 0.77
* vectorWhile regressed by 0.91
* Overall benchmarks improved by 1.0

Please find your performance results at https://rir-benchmarks.prl.fit.cvut.cz/diff?job_ids[]=1429874482&job_ids[]=1415143749&selection=all

skrynski committed:
minor change

skrynski committed:
fix

skrynski committed:
ldconst  deopt reason

o- commented:
Here are some stats for your PR:

* WARNING: the longest CI job tests_debug2 took 2.2h
* real_thing suite improved by 1.05
* nbody_naive regressed by 0.79
* spectralnorm regressed by 0.91
* reversecomplement_2 regressed by 0.95
* vectorWhile regressed by 0.91
* Overall benchmarks improved by 1.01

Please find your performance results at https://rir-benchmarks.prl.fit.cvut.cz/diff?job_ids[]=1439009355&job_ids[]=1432940244&selection=all

skrynski committed:
avoid unnecessary Not/Assume Not

o- commented:
Here are some stats for your PR:

* WARNING: the longest CI job tests_debug2 took 1.9h
* real_thing suite improved by 1.04
* simple suite improved by 1.02
* Bounce_nonames_simple regressed by 0.95
* convolution_v regressed by 0.88
* nbody_naive regressed by 0.78
* fasta_2 regressed by 0.9
* reversecomplement_naive regressed by 0.93
* spectralnorm regressed by 0.93
* mandelbrot_naive_ascii regressed by 0.94
* vectorWhile regressed by 0.9
* Overall benchmarks improved by 1.0

Please find your performance results at https://rir-benchmarks.prl.fit.cvut.cz/diff?job_ids[]=1444153669&job_ids[]=1432940244&selection=all

skrynski committed:
try something

skrynski committed:
fix

skrynski committed:
deopt reason

skrynski committed:
fix gc torture?

skrynski committed:
using opcode() instead of calculating offset

skrynski committed:
change opcode() pointer arithmetics

skrynski committed:
fix srcCode bug

o- commented:
Here are some stats for your PR:

* WARNING: the longest CI job tests_debug2 took 2.23h
* real_thing suite improved by 1.06
* simple suite improved by 1.03
* nbody_naive regressed by 0.77
* vectorWhile regressed by 0.91
* Overall benchmarks improved by 1.01

Please find your performance results at https://rir-benchmarks.prl.fit.cvut.cz/diff?job_ids[]=1447818311&job_ids[]=1432940244&selection=all

skrynski committed:
Merge branch 'master' into assumeNotBranch-rir2pir

o- commented:
Here are some stats for your PR:

* WARNING: the longest CI job test_features_3 took 1.92h
* Bounce_nonames regressed by 0.95
* nbody_naive regressed by 0.75
* vectorWhile regressed by 0.94
* Overall benchmarks regressed by 1.0

Please find your performance results at https://rir-benchmarks.prl.fit.cvut.cz/diff?job_ids[]=1451992630&job_ids[]=1450318169&selection=all

o- committed:
cleanup to understand what is going on

I really could not understand the tripple-negate logic that was put in
place here. Decided to refactor it, so I can finally understand again
what is going on. No guarantee that this does the same thing as before.
But I think it does what it should do...

o- committed:
re-add casting type-checked values

skrynski committed:
suppress variableScope check

skrynski committed:
cppcheck again

o- commented:
Here are some stats for your PR:

* WARNING: the longest CI job test_features_3 took 2.57h
* real_thing suite regressed by 0.97
* convolution regressed by 0.87
* nbody_naive regressed by 0.76
* spectralnorm regressed by 0.94
* scalarFor regressed by 0.78
* Overall benchmarks regressed by 0.99

Please find your performance results at https://rir-benchmarks.prl.fit.cvut.cz/diff?job_ids[]=1458707522&job_ids[]=1450318169&selection=all

skrynski committed:
ElideEnvSpec reports changes

o- commented:
Here are some stats for your PR:

* WARNING: the longest CI job tests_debug2 took 1.95h
* real_thing suite regressed by 0.97
* convolution regressed by 0.87
* spectralnorm regressed by 0.94
* fasta_naive_2 regressed by 0.95
* scalarFor regressed by 0.84
* vectorWhile regressed by 0.94
* Overall benchmarks regressed by 1.0

Please find your performance results at https://rir-benchmarks.prl.fit.cvut.cz/diff?job_ids[]=1507549755&job_ids[]=1450318169&selection=all

skrynski committed:
Merge branch 'master' into assumeNotBranch-rir2pir

o- commented:
Here are some stats for your PR:

* WARNING: the longest CI job test_release_2 took 2.0h
* Bounce_nonames regressed by 0.83
* convolution_v regressed by 0.89
* reversecomplement_naive regressed by 0.91
* fasta_3 regressed by 0.94
* emptyFor regressed by 0.9
* vectorWhile regressed by 0.93
* Overall benchmarks regressed by 1.0

Please find your performance results at https://rir-benchmarks.prl.fit.cvut.cz/diff?job_ids[]=1509062472&job_ids[]=1504675376&selection=all

skrynski committed:
refactoring typefeedback

o- commented:
Here are some stats for your PR:

* WARNING: the longest CI job test_release_2 took 2.46h
* Bounce_nonames regressed by 0.9
* mandelbrot_naive_ascii regressed by 0.92
* fannkuchredux regressed by 0.94
* scalarFor regressed by 0.9
* vectorWhile regressed by 0.93
* Overall benchmarks regressed by 1.0

Please find your performance results at https://rir-benchmarks.prl.fit.cvut.cz/diff?job_ids[]=1516031745&job_ids[]=1504675376&selection=all

o- committed:
cleanup deopt reason recording

Capture the deopt reason already in the assume instruction.

Cleanup and reafactoring.

Pair-programming with @skrynski

skrynski committed:
fix initialization of reason in assume

o- commented:
Here are some stats for your PR:

* WARNING: the longest CI job test_release_2 took 2.22h
* are-we-fast-r suite regressed by 0.98
* real_thing suite regressed by 0.93
* Bounce_nonames_simple regressed by 0.94
* convolution regressed by 0.81
* convolution_v regressed by 0.89
* scalarFor regressed by 0.89
* vectorWhile regressed by 0.91
* Overall benchmarks regressed by 0.99

Please find your performance results at https://rir-benchmarks.prl.fit.cvut.cz/diff?job_ids[]=1530063417&job_ids[]=1523649355&selection=all

skrynski committed:
run again

o- commented:
Here are some stats for your PR:

* WARNING: the longest CI job test_release_2 took 2.24h
* real_thing suite regressed by 0.94
* Bounce_nonames_simple regressed by 0.94
* convolution regressed by 0.81
* convolution_v regressed by 0.89
* nbody_naive regressed by 0.93
* vectorWhile regressed by 0.91
* Overall benchmarks regressed by 1.0

Please find your performance results at https://rir-benchmarks.prl.fit.cvut.cz/diff?job_ids[]=1531069660&job_ids[]=1523649355&selection=all

skrynski committed:
remove old comments

o- commented:
Here are some stats for your PR:

* WARNING: the longest CI job test_release_2 took 1.98h
* are-we-fast-r suite regressed by 0.98
* real_thing suite regressed by 0.96
* Bounce_nonames_simple regressed by 0.91
* convolution regressed by 0.8
* mandelbrot_naive_ascii regressed by 0.84
* spectralnorm regressed by 0.95
* vectorWhile regressed by 0.91
* Overall benchmarks regressed by 1.0

Please find your performance results at https://rir-benchmarks.prl.fit.cvut.cz/diff?job_ids[]=1532890726&job_ids[]=1523649355&selection=all
